### ISSUE TITLE
gt_mask for DBNet Training

### ISSUE THREAD

**Issue by abdksyed** [open-mmlab/mmocr#537]
Hi,

I am trying to train DBNet on custom data for text-detection. I have bounding boxes annotated, but don't have the segmentation mask. 

I tried to use the DBNet Resnet 50 config file to create similar file for my custom data, but the `dict(type='EastRandomCrop', target_size=(640, 640)),` is throwing error.

```
File "mmocr/mmocr/datasets/pipelines/dbnet_transforms.py", line 150, in __call__
img, results['gt_masks']
``` 

It's understandable that since I don't have segmentation annotations, it can't work. I wanted to know does DBNet has a hard requirement for segmentation mask, and can't it be trained only using bounding boxes.

If segmentation mask is not hard requirement, than can we comment out EastRandomCrop from the config, and what would be the alternate pipeline item for cropping image instead of EastRandomCrop.

**Comment by abdksyed** [comment]
ohh!!!, getting segmentation annotations will be very difficult.

I am using it on documents detecting words, and each document has around 300-500 boxes, doing segmentation annotation will be really difficult. I shall try converting BBox to Segmentation mask itself maybe. Any idea(hope) that this may work out???

**Comment by gaotongxiao** [comment]
Actually it's not difficult at all. You just need to fill the segmentation field with coordinates of 4 vertices, which should be exactly the same as your bounding box. Here is a reference:
 https://github.com/open-mmlab/mmocr/blob/b8f7ead74cb0200ad5c422e82724ca6b2eb1c543/tools/data/textdet/icdar_converter.py#L99-L126

**Comment by gaotongxiao** [comment]
You might need to check if your training annotation also has the "segmentation" key added.

**Comment by abdksyed** [comment]
oh yeah, In the config file, before I turned with_mask as False, when I was not having segmentation mask. I forgot about it totally after adding segmentation mask. Thanks.

I added it back in the config, looks like it's training. Thanks a lot.

One more question, I see that it's having max_epochs as 1200. How do I change that, couldn't find anywhere in config.

```
2021-10-21 07:08:36,223 - mmocr - INFO - workflow: [('train', 1)], max: 1200 epochs
```

**Comment by abdksyed** [comment]
Thanks a lot. 

It's training, will have to see the results at the end. Currently after 300 epochs this is the status. Since the model was already good when I tested on my data with alread pre-trained model it had

`{'hmean-iou:recall': 0.8750361585189471, 'hmean-iou:precision': 0.9279141104294478, 'hmean-iou:hmean': 0.900699717135626}`

```
2021-10-21 09:19:21,273 - mmocr - INFO - thr 0.3, recall：0.959, precision: 0.944, hmean:0.952
2021-10-21 09:22:00,206 - mmocr - INFO - thr 0.4, recall：0.959, precision: 0.949, hmean:0.954
2021-10-21 09:24:38,491 - mmocr - INFO - thr 0.5, recall：0.959, precision: 0.953, hmean:0.956
2021-10-21 09:27:16,682 - mmocr - INFO - thr 0.6, recall：0.957, precision: 0.955, hmean:0.956
2021-10-21 09:29:53,625 - mmocr - INFO - thr 0.7, recall：0.951, precision: 0.955, hmean:0.953
2021-10-21 09:32:06,888 - mmocr - INFO - thr 0.8, recall：0.813, precision: 0.963, hmean:0.882
2021-10-21 09:32:09,435 - mmocr - INFO - thr 0.9, recall：0.023, precision: 0.941, hmean:0.045

2021-10-21 09:32:09,436 - mmocr - INFO - Epoch(val) [300][15]	
hmean-iou:recall: 0.9572, hmean-iou:precision: 0.9547, hmean-iou:hmean: 0.9559
```

Will close this issue now. Thanks a lot!!!
